qiskit/algorithms/optimizers/suffix_averaging.py

```diff
@@ -0,0 +1,163 @@
+                is_accepted: bool,
+            ):
+                self._circ_params.append(x_next)
+                if original_callback is not None:
+                    original_callback(nfev, x_next, fx_next, stepsize, is_accepted)
+
+        elif isinstance(self._optimizer, GradientDescent):
+
+            def callback(nfev: int, x: POINT, fun: float, stepsize: float):
+                self._circ_params.append(x)
+                if original_callback is not None:
+                    original_callback(nfev, x, fun, stepsize)
+
+        else:
+
+            def callback(x):
+                self._circ_params.append(x)
+                if original_callback is not None:
+                    return original_callback(x)
+                return None
+
+        self._optimizer.callback = callback
+        super().__init__()
+
+    def get_support_level(self):
+        """Return support level dictionary"""
+        return self._optimizer.get_support_level()
+
+    def _return_suffix_average(self) -> np.ndarray:
+
+        if isinstance(self._optimizer, NFT):
+            n_params = 1 if isinstance(self._circ_params[0], float) else len(self._circ_params[0])
+            n_iterates = int(len(self._circ_params))
+            n_repitition = int(len(self._circ_params) / n_params)
+            if n_repitition < self._n_params_suffix:
+                warnings.warn("The total number of iterations is less than n_params_suffix.")
+                averaged_param = np.zeros_like(self._circ_params[0])
+                for j in range(n_repitition):
+                    averaged_param += self._circ_params[n_iterates - n_params * j - 1]
+                averaged_param /= self._n_params_suffix
+                return averaged_param
+
+            averaged_param = np.zeros_like(self._circ_params[0])
+            for j in range(self._n_params_suffix):
+                averaged_param += self._circ_params[n_iterates - n_params * j - 1]
+            averaged_param /= self._n_params_suffix
+
+        else:
+            n_iterates = len(self._circ_params)
+
+            if n_iterates < self._n_params_suffix:
+                warnings.warn("The total number of iterations is less than n_params_suffix.")
+                return np.average(np.asarray(self._circ_params), axis=0)
+
+            averaged_param = np.zeros_like(self._circ_params[0])
+            for j in range(self._n_params_suffix):
+                averaged_param += self._circ_params[n_iterates - j - 1]
+            averaged_param /= self._n_params_suffix
+
+        return averaged_param
+
+    def minimize(
+        self,
+        fun: Callable[[POINT], float],
+        x0: POINT,
+        jac: Optional[Callable[[POINT], POINT]] = None,
+        bounds: Optional[List[Tuple[float, float]]] = None,
+    ) -> OptimizerResult:
+
+        result = self._optimizer.minimize(fun, x0, jac=jac, bounds=bounds)
+        result.x = self._return_suffix_average()
+        result.fun = fun(np.copy(result.x))
```

Comment:
Why is the copy here needed?

Reply:
Oh, I removed it. Thank you for pointing that out.